examples/energy-management-app/energy-management-common/water-heater/src/WhmDelegateImpl.cpp

```diff
@@ -273,61 +300,83 @@ Status WaterHeaterManagementDelegate::HandleCancelBoost()
  *
  *********************************************************************************/
 
-void WaterHeaterManagementDelegate::SetWaterTemperature(uint16_t waterTemperature)
+int16_t WaterHeaterManagementDelegate::GetActiveTargetWaterTemperature() const
+{
+    // Determine the target temperature. If a boost command is in progress and has a mBoostTemporarySetpoint value use that as the
+    // target temperature.
+    // Note, in practise the actual heating is likely to be controlled by the thermostat's occupiedHeatingSetpoint most of the
+    // time, and the TemporarySetpoint (if not null) would be overiding the thermostat's occupiedHeatingSetpoint.
+    // However, this code doesn't rely upon the thermostat cluster.
+    int16_t targetTemperature = (mBoostState == BoostStateEnum::kActive && mBoostTemporarySetpoint.HasValue())
+        ? static_cast<int16_t>(mBoostTemporarySetpoint.Value())
```

Comment:
```suggestion
        ? mBoostTemporarySetpoint.Value()
```
It's already the right type, no?